ResearchKit/Common/ORKSurveyAnswerCellForText.m

```diff
@@ -244,7 +245,18 @@ - (void)prepareView {
 }
 
 - (BOOL)shouldContinue {
-    return ![self correctValueIfNeeded];
+    ORKTextAnswerFormat *answerFormat = (ORKTextAnswerFormat *)[self.step impliedAnswerFormat];
+    if (answerFormat.isEmail) {
+        BOOL isValid = [answerFormat isAnswerValidWithString:self.answer];
+
+        if (!isValid){
+            [self showValidityAlertWithMessage:self.answer];
+        }
+        
+        return isValid;
+    } else {
+        return ![self correctValueIfNeeded];
```

Comment:
Do you think we should drop `correctValueIfNeeded` and use alert message for text content exceeding length limit? Just cut the text from the end seems too arbitrary.